lisa/datautils.py

```diff
@@ -504,6 +504,40 @@ def df_window(df, window, method='inclusive', clip_window=False):
     """
     return _data_window(df, window, method, clip_window)
 
+def _data_adjacent_index(data, ref, prev):
+    """
+    ``data`` can either be a :class:`pandas.DataFrame` or :class:`pandas.Series`.
+    """
+    res = None
+
+    if prev and ref > data.index[0]:
+        res = data.index.get_loc(ref, "ffill")
+    elif ref < data.index[-1]:
+        res = data.index.get_loc(ref, "bfill")
+
+    return res
+
+def df_adjacent_index(df, ref, prev=True):
+    """
+    :param ref: The index value to look around
+
+    :param prev: Whether to find the previous index (``True``)
```

Comment:
Could we use `method="prev"` and `method="next"` instead of a boolean, and remove the default value ?
That avoids putting some arbitrary emphasis on one or the other, and makes the client code clear as to which behavior is in use. Since the name of the function is neutral (as it should), there is no reason for it to privilege prev over next.

Reply:
I went with that, then backtracked because it was a boolean information. Let's backtrack again!